src/main/java/com/ericsson/ei/controller/SubscriptionController.java

```diff
@@ -33,36 +34,36 @@
      * 
      */
     @RequestMapping(value = "", method = RequestMethod.POST)
-    public ResponseEntity<?> createSubscription(
+    public ResponseEntity<List<com.ericsson.ei.controller.model.SubscriptionResponse>> createSubscription(
         @Valid
         @RequestBody
         List<com.ericsson.ei.controller.model.Subscription> subscription);
 
     /**
-     * Modify an existing Subscription.
+     * Modify an existing Subscriptions.
      * 
      */
     @RequestMapping(value = "", method = RequestMethod.PUT)
-    public ResponseEntity<com.ericsson.ei.controller.model.SubscriptionResponse> updateSubscriptions(
+    public ResponseEntity<List<com.ericsson.ei.controller.model.SubscriptionResponse>> updateSubscriptions(
         @Valid
         @RequestBody
         List<com.ericsson.ei.controller.model.Subscription> subscription);
 
     /**
-     * Returns the subscription rules for given subscription name.
+     * Returns the subscriptions for given subscription names.
      * 
      */
     @RequestMapping(value = "/{subscriptionName}", method = RequestMethod.GET)
-    public ResponseEntity<List<com.ericsson.ei.controller.model.Subscription>> getSubscriptionById(
+    public ResponseEntity<GetSubscriptionResponse> getSubscriptionById(
         @PathVariable(required = false)
         String subscriptionName);
 
     /**
-     * Removes the subscription from the database.
+     * Removes the subscriptions from the database for given subscription names.
      * 
      */
     @RequestMapping(value = "/{subscriptionName}", method = RequestMethod.DELETE)
-    public ResponseEntity<com.ericsson.ei.controller.model.SubscriptionResponse> deleteSubscriptionById(
+    public ResponseEntity<List<com.ericsson.ei.controller.model.SubscriptionResponse>> deleteSubscriptionById(
         @PathVariable(required = false)
```

Comment:
If going to return a List of failed Deletion of subscriptions -> we need also be able to take input "SubscriptionName" as a list also -> e.g. List<String> subscriptionNameList. And iterate and delete.
Or we maybe keep this entry point and add a new DELETE Request entrypoint "/" which takes a list of subscriptionNames.

Reply:
I tried to generate in RAML delete endpoint, that handle list in request body, but failed with it. So now it is implemented in this endpoint. It can handle one or several subscription names comma seperated.